pkg/controller/bundle_sync_task.go

```diff
@@ -199,106 +281,128 @@ func (st *bundleSyncTask) handleProcessResult(retriable bool, processErr error)
 		return false, processErr
 	}
 
-	// Construct resource conditions and check if there were any resource errors
-	resourceStatuses := make([]smith_v1.ResourceStatus, 0, len(st.processedResources))
-	var failedResources []smith_v1.ResourceName
-	retriableResourceErr := true
-	statusUpdated := false
-	for _, res := range st.bundle.Spec.Resources { // Deterministic iteration order
-		blockedCond := smith_v1.ResourceCondition{Type: smith_v1.ResourceBlocked, Status: smith_v1.ConditionFalse}
-		inProgressCond := smith_v1.ResourceCondition{Type: smith_v1.ResourceInProgress, Status: smith_v1.ConditionFalse}
-		readyCond := smith_v1.ResourceCondition{Type: smith_v1.ResourceReady, Status: smith_v1.ConditionFalse}
-		errorCond := smith_v1.ResourceCondition{Type: smith_v1.ResourceError, Status: smith_v1.ConditionFalse}
-
-		if resInfo, ok := st.processedResources[res.Name]; ok {
-			// Resource was processed
-			switch resStatus := resInfo.status.(type) {
-			case resourceStatusDependenciesNotReady:
-				blockedCond.Status = smith_v1.ConditionTrue
-				blockedCond.Reason = smith_v1.ResourceReasonDependenciesNotReady
-				blockedCond.Message = fmt.Sprintf("Not ready: %q", resStatus.dependencies)
-			case resourceStatusInProgress:
-				inProgressCond.Status = smith_v1.ConditionTrue
-			case resourceStatusReady:
-				readyCond.Status = smith_v1.ConditionTrue
-			case resourceStatusError:
-				errorCond.Status = smith_v1.ConditionTrue
-				errorCond.Message = resStatus.err.Error()
-				if resStatus.isRetriableError {
-					errorCond.Reason = smith_v1.ResourceReasonRetriableError
+	bundleUpdated := false
+
+	if st.newFinalizers != nil {
+		// Update finalizers
+		st.bundle.Finalizers = st.newFinalizers
+		// Set status to "InProgress: True"
+		// (there will be one more iteration for resource sync)
+		// TODO: add more details to the status (reason, resources status)?
+		st.bundle.Status = smith_v1.BundleStatus{
+			Conditions: []smith_v1.BundleCondition{
+				{Type: smith_v1.BundleInProgress, Status: smith_v1.ConditionTrue},
```

Comment:
@ash2k suggestion is this shouldn't change the status since finalizer changes don't necessarily correspond to operations in progress.

Reply:
@nilebox status should only be set here if `DeletionTimestamp` is set.

Reply:
```go
if st.newFinalizers != nil {
  // set finalizers
}
if st.bundle.DeletionTimestamp == nil {
   // bla bla
} else {
   // set in progress
}
```
WDYT?

Reply:
@ash2k @amckague status should change to "InProgress" regardless of `DeletionTimestamp`. Example: user sends an update request which removes finalizers and changes resource list at the same time. Hence the "ready" status which was set before this update is not up-to-date anymore.

Reply:
@amckague so yes, you're right that in reality finalizers change don't necessarily need any updates of the resources, but since we don't have any data about the potential issue I described above, we need to set it to "InProgress" to be safe. The next iteration will correct this status back to "Ready: True" if there's nothing to do.

Reply:
>  Example: user sends an update request which removes finalizers and changes resource list at the same time. Hence the "ready" status which was set before this update is not up-to-date anymore.

@nilebox it will still be not up to date if you set it to anything else than what it should actually be. You are trying to solve the problem that should be solved using generation+observedGeneration (https://github.com/atlassian/smith/issues/102). Also, status may be out of sync with spec [by design](https://github.com/kubernetes/community/blob/master/contributors/devel/api-conventions.md#spec-and-status):

> When a new version of an object is POSTed or PUT, the "spec" is updated and available immediately. Over time the system will work to bring the "status" into line with the "spec".

Another thing - this code is overwriting the whole status block. It should not. It is potentially erasing useful information about the state of the resource.

Reply:
@ash2k the reason I had to add this status update was a failing `TestUpdate` integration test.
The `AssertBundleTimeout()` method waits until the status is `Ready` and then goes to check if resources are meeting expectations. It turns out that Smith just updated the finalizer and didn't process the resource changes, hence the failing test. Once I introduced an `InProgress` status for finalizer update, the test succeeded. The fix seemed to be appropriate.
Do you think that we need to rewrite the test to resolve this issue instead?

Reply:
Leaving a TODO that we might fix this issue differently in future (using `ObservedGeneration` or something else).